packages/thirdweb/src/extensions/farcaster/read/getNonce.ts

```diff
@@ -0,0 +1,47 @@
+import type { ThirdwebClient } from "../../../client/client.js";
+import { withCache } from "../../../utils/promise/withCache.js";
+import type { Chain } from "../../../chains/types.js";
+import { getKeyGateway } from "../contracts/getKeyGateway.js";
+import { nonces } from "../__generated__/IKeyGateway/read/nonces.js";
+import type { Address } from "abitype";
+
+export type GetNonceParams = {
+  client: ThirdwebClient;
+  address: Address;
+  chain?: Chain;
+  disableCache?: boolean;
+};
+
+/**
+ * Retrieves the current key gateway nonce for an account.
+ * @param options - Parameters to pass to the function.
+ * @returns A promise that resolves to the current nonce.
+ * @extension FARCASTER
+ * @example
+ * ```ts
+ * import { getNonce } from "thirdweb/extensions/farcaster";
+ *
+ * const nonce = await getNonce({
+ *  client,
+ * 	address,
+ * });
+ * ```
+ */
+export async function getNonce(options: GetNonceParams): Promise<bigint> {
+  const fetch = () => {
+    const contract = getKeyGateway({
+      client: options.client,
+      chain: options.chain,
+    });
+    return nonces({ contract, account: options.address });
+  };
+
+  if (options.disableCache) {
+    return fetch();
+  }
+
+  return withCache(fetch, {
+    cacheKey: `${options.address}:getNonce`,
+    cacheTime: 5 * 60 * 1000, // 5 minutes
+  });
```

Comment:
I *may* be wrong but i believe instead of doing it this way you might be able to set `cacheTime: 0` for the "disableCache"case. which means you wouldn't have to fork your logic and could define your fetch in-line

same thing for the other calls with this pattern

Reply:
Oh sweet, will update